event_model/schemas/run_start.json

```diff
@@ -4,6 +4,34 @@
             "title" : "data_type",
             "patternProperties": {"^([^./]+)$": {"$ref": "#/definitions/data_type"}},
             "additionalProperties": false
+        },
+        "data_mapping": {
+            "description": "Where to get the data from",
+            "type": "object",
+            "properties" : {
+                "stream": {"type": "string"},
+                "location": {"enum" : ["event", "configuration"]},
+                "field": {"type": "string"}
+            },
+            "required" : ["stream", "location", "field"],
+            "additionalProperties": false
+        },
+        "technique": {
```

Comment:
The term "technique" might be too limiting. This provides a generic mechanism for mapping any externally-defined metadata schema to the contents of the documents to come. Those schemas might be broken up by experimental technique, by downstream analysis process (applicable to more than one technique such as "scattering" and "diffraction"), by institution, by domain, etc.

Reply:
"data_remapping", "datamap",  "DateMap", "Application", "application_definitions"?

I think we have a bunch of helper functions like `list_applications(h)`, `iter_applications(h, name=None)` that yields dicts (?) full of {base types or xarrays}?